awslabs/aws-lambda-powertools-typescript
# chore(ci): revisit stale issues automation

## Description of your changes

As described in #1340, the current automation for stale issues was broken. While in the issue I incorrectly assumed that this was [stalebot](https://github.com/probot/stale)'s fault, this was not the case.

The repo actually had two different automations for the same purpose: one was the aforementioned stalebot and the other was a workflow called [`stale-issues.yml`](https://github.com/awslabs/aws-lambda-powertools-typescript/blob/main/.github/workflows/stale-issues.yml) that is run every day at midnight. The latter was the one that was broken.

Additionally, while looking into stalebot, I found out that it's now unmaintained and deprecated, so in this PR I am removing its configuration from the repo.

The stale workflow was previously running using the [`aws-actions/stale-issue-cleanup`](https://github.com/aws-actions/stale-issue-cleanup) action, however after comparing it with the official [`actions/stale`](https://github.com/actions/stale) the differences were minimal and I decided to go with the official one.

As mentioned, the workflow is expected to run once a day at midnight. It's not clear which timezone this refers to, given that the cron feature [doesn't support timezones](https://github.com/community/community/discussions/13454), however based on previous executions this seems to be between midnight and 1AM CET, which is good enough.

The workflow operates only on issues and only on issues explicitly labeled as `need-more-information` and `status/discussing`. This should limit the amount of issues closed automatically given that maintainers should use these two labels in conjunction to denote that an issue is being discussed but more information is needed from the person who opened it. 

If no interaction is made (aka the info that was missing is not provided), the workflow will mark the issue as stale and apply the `status/pending-close-response-required` label. The workflow will also leave a comment that says:
> This issue has not received a response in 2 weeks. If you still think there is a problem, please leave a comment to avoid the issue from automatically closing.

If 7 more days pass, the issue will finally be closed as `not_planned` (close reason) and labeled as `status/rejected`, with a comment:
> Greetings! We are closing this issue because it has been open a long time and hasn’t been updated in a while and may not be getting the attention it deserves. We encourage you to check if this is still an issue in the latest release and if you find that this is still a problem, please feel free to comment or reopen the issue.

If instead the issue is interacted with within a week, the label will be removed and the issue stops being stale.

Issues that haven't yet been triaged (labeled as `triage`) or that have a clear status as indicated by the labels: `status/confirmed`, `status/blocked`, `status/on-hold`, `status/completed`, are excluded by the automation and left open.

PRs should be excluded by this workflow. The documentation around this is a bit unclear, but based [on a comment](https://github.com/actions/stale/issues/688) in the project's issue it seems that setting a negative amount of days to the PR config of the action avoids marking any PR as stale. 

### How to verify this change

Once merged, check that there are no issues in the workflow syntax. Then a day after check the workflow execution to see that there were no runtime errors.

### Related issues, RFCs



**Issue number:** #1340

## Checklist


### Breaking change checklist

***Is it a breaking change?:*** NO


---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
